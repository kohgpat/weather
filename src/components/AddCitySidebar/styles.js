import styled from "styled-components";
import ButtonComponent from "../Button";

export const Sidebar = styled.div`
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  width: 300px;
  background-color: #fff;
  display: none;
  z-index: 1;
  border-left: var(--add-city-sidebar-border);

  ${props => props.isVisible && `
    display: flex;
    flex-direction: column;
  `}
`;

export const SidebarHeader = styled.header`
  display: flex;
  padding: 20px;
`;

export const SidebarCloseButton = styled(ButtonComponent).attrs({ type: "button" })`
  width: auto;
  margin-left: auto;
`;
