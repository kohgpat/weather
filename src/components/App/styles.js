import styled from "styled-components";

export const App = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
`;
