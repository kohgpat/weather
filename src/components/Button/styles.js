import styled from "styled-components";

export const Button = styled.button`
  font-size: 14px;
  font-weight: bold;
  font-family: "Roboto Condensed", "Helvetica Neue", "Helvetica", Arial, sans-serif;
  padding: 8px 16px;
  height: 40px;
  border: 0;
  background-color: #eee;
  width: 100%;
`;
