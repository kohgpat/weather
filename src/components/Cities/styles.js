import styled from "styled-components";

export const Cities = styled.div`
  margin-top: calc(50px + var(--topbar-height));
  padding: 20px;

  @supports (display: grid) {
    display: grid;
    grid-template-columns: repeat(1, 1fr);
    grid-gap: 30px;
    grid-auto-rows: minmax(300px, auto);

    @media (min-width: 768px) {
      grid-template-columns: repeat(2, 1fr);
    }

    @media (min-width: 1028px) {
      grid-template-columns: repeat(3, 1fr);
    }
  }

  @supports not (display: grid) {
    display: flex;
    flex-direction: row;
    flex-shrink: 0;
    align-items: center;
    justify-content: center;
    width: 100%;
    flex-wrap: wrap;
  }
`;

export const NoCitiesMessage = styled.div`
  margin-top: calc(50px + var(--topbar-height));
  padding: 20px;
`;
