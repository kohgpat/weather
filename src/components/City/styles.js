import styled from "styled-components";

export const City = styled.div`
  height: 280px;
  width: 280px;
  background-color: #fff;
  border-radius: 4px;
  box-shadow: 0 0 20px -2px rgba(0,0,0,0.1);
  display: flex;
  flex-direction: column;
  padding: 10px;

  @media (min-width: 768px) {
    height: 300px;
    width: 300px;
  }

  @supports not (display: grid) {
    margin-top: 30px;

    @media (orientation: landscape) {
      &:not(:last-child) {
        margin-right: 30px;
      }
    }

    @media (min-width: 768px) {
      &:not(:last-child) {
        margin-right: 30px;
      }
    }
  }
`;

export const Header = styled.header`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const Name = styled.h1`
  margin: 10px;
  width: 80%;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
`;

export const RemoveButton = styled.button.attrs({ type: "button" })`
  padding: 0;
  font-weight: bold;
  font-size: 32px;
  background-color: transparent;
  border: 0;
  margin: 10px 10px 10px auto;
  visibility: visibility;
  cursor: pointer;

  @media (min-width: 768px) {
    visibility: hidden;

    ${City}:hover & {
      visibility: visible;
    }
  }
`;

export const WeatherIcon = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
  margin: auto 0;
`;

export const Stats = styled.div`
  margin: 10px;
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

export const Stat = styled.h4`
  text-transform: capitalize;
  margin: 0;

  ${props =>
    props.major &&
    `
    font-size: 3rem;
  `};
`;
