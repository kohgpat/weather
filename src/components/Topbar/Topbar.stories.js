import React from "react";
import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";
import Topbar from "./index";

storiesOf("Topbar", module).add("default", () => (
  <Topbar
    settings={{
      units: "celcius"
    }}
    toggleSettingsUnits={() => action("toggleSettingsUnits")}
  />
));
