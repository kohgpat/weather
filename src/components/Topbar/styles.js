import styled from "styled-components";
import ContainerComponent from "../Container";
import ButtonComponent from "../Button";

export const Topbar = styled.div`
  position: fixed;
  background-color: var(--topbar-background-color);
  width: var(--topbar-width);
  height: var(--topbar-height);
  flex-shrink: 0;
  padding: 0 20px;
  box-shadow: 0 4px 2px -2px rgba(0,0,0,0.1);

  ${props =>
    props.isMobile &&
    props.addCitySidebar &&
    props.addCitySidebar.isVisible &&
    `
    display: none;
  `};
`;

export const Container = styled(ContainerComponent)`
  position: relative;
  align-items: center;
  margin: 0 auto;
  justify-content: space-around;
  height: 100%;
`;

export const Settings = styled.div`
  position: absolute;
  left: 0;
`;

export const SettingsControl = styled(ButtonComponent).attrs({
  type: "button"
})`
  width: auto;
  margin: 0;
  background-color: var(--topbar-settings-control-background-color);
  color: var(--topbar-settings-control-color);

  ${props =>
    props.isActive &&
    `
    background-color: var(--topbar-settings-control-background-color-active);
    color: var(--topbar-settings-control-color-active);
  `};
`;

export const Name = styled.h1`
  font-family: var(--topbar-name-font-family);
  font-size: 24px;

  @media (min-width: 768px) {
    font-size: 32px;
  }
`;

export const AddCitySidebarToggle = styled(ButtonComponent).attrs({
  type: "button"
})`
  width: auto;
  position: absolute;
  right: 0;
`;
