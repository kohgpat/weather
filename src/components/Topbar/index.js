import React, { Component } from "react";
import * as styles from "./styles";

class Topbar extends Component {
  state = {
    isMobile: this.props.isMobile
  };

  render() {
    const {
      settings,
      isMobile,
      addCitySidebar,
      toggleSettingsUnits,
      toggleAddCitySidebar
    } = this.props;

    return (
      <styles.Topbar isMobile={isMobile} addCitySidebar={addCitySidebar}>
        <styles.Container>
          <styles.Settings>
            <styles.SettingsControl
              isActive={settings.units === "celcius"}
              disabled={settings.units === "celcius"}
              onClick={() => toggleSettingsUnits("celcius")}>
              {isMobile ? "C" : "Celcius"}
            </styles.SettingsControl>

            <styles.SettingsControl
              isActive={settings.units === "fahrenheit"}
              disabled={settings.units === "fahrenheit"}
              onClick={() => toggleSettingsUnits("fahrenheit")}>
              {isMobile ? "F" : "Fahrenheit"}
            </styles.SettingsControl>
          </styles.Settings>

          <styles.Name>Weather</styles.Name>
          <styles.AddCitySidebarToggle onClick={toggleAddCitySidebar}>
            +
          </styles.AddCitySidebarToggle>
        </styles.Container>
      </styles.Topbar>
    );
  }
}

export default Topbar;
