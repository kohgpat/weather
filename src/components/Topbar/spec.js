import React from "react";
import Topbar from "./index";
import * as styles from "./styles";
import { shallow, mount } from "enzyme";

const settings = {
  units: "celcius"
};
const toggleSettingsUnits = jest.fn();
const toggleAddCitySidebar = jest.fn();

it("render", () => {
  const topbar = shallow(
    <Topbar
      settings={settings}
      toggleSettingsUnits={toggleSettingsUnits}
      toggleAddCitySidebar={toggleAddCitySidebar}
    />
  );
  expect(topbar).toMatchSnapshot();
});

it("renders text and data", () => {
  const topbar = mount(
    <Topbar
      settings={settings}
      toggleSettingsUnits={toggleSettingsUnits}
      toggleAddCitySidebar={toggleAddCitySidebar}
    />
  );
  const name = topbar.find(styles.Name);
  expect(name.text()).toEqual("Weather");
});
