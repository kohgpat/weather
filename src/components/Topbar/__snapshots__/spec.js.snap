// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<styled.div>
  <Styled(Container)>
    <styled.div>
      <Styled(Button)
        disabled={true}
        isActive={true}
        onClick={[Function]}
      >
        Celcius
      </Styled(Button)>
      <Styled(Button)
        disabled={false}
        isActive={false}
        onClick={[Function]}
      >
        Fahrenheit
      </Styled(Button)>
    </styled.div>
    <styled.h1>
      Weather
    </styled.h1>
    <Styled(Button)
      onClick={[MockFunction]}
    >
      +
    </Styled(Button)>
  </Styled(Container)>
</styled.div>
`;
